modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParameterResolverUtils.java

```diff
@@ -0,0 +1,212 @@
+import static org.mule.runtime.module.extension.internal.runtime.resolver.ParametersResolver.fromValues;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.meta.model.parameter.ParameterGroupModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.el.ExpressionManager;
+import org.mule.runtime.module.extension.api.loader.java.type.Type;
+import org.mule.runtime.module.extension.internal.loader.java.property.ParameterGroupModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.type.property.ExtensionParameterDescriptorModelProperty;
+import org.mule.runtime.module.extension.internal.runtime.objectbuilder.DefaultObjectBuilder;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utils class to create {@link ParametersResolver}.
+ *
+ * @since 4.5
+ */
+public class ParameterResolverUtils {
+
+  /**
+   * Creates a {@link ParametersResolver} of a {@link ParameterizedModel} based on static values of its parameters.
+   *
+   * @param parameters         static parameter values.
+   * @param parameterizedModel the parameterized group.
+   * @param muleContext        the mule context.
+   * @param disableValidations whether validations should be disabled or not.
+   * @param reflectionCache    a reflection cache.
+   * @param expressionManager  the expression manager.
+   * @param parametersOwner    the owner of the parameters from the parameters resolver.
+   * @return the corresponding {@link ParametersResolver}
+   * @throws MuleException
+   */
+  public static ParametersResolver getParameterResolverForParameterizedModelFromStaticValues(Map<String, Object> parameters,
+                                                                                             ParameterizedModel parameterizedModel,
+                                                                                             MuleContext muleContext,
+                                                                                             boolean disableValidations,
+                                                                                             ReflectionCache reflectionCache,
+                                                                                             ExpressionManager expressionManager,
+                                                                                             String parametersOwner)
+      throws MuleException {
+    Map<String, ValueResolver> resolvers = new HashMap<>();
+
+    for (ParameterGroupModel parameterGroupModel : parameterizedModel.getParameterGroupModels()) {
```

Comment:
where is all this code taken from? What I would expect here is for this code to have been moved from other places in the SDK in which we are already generating parameter resolvers. I would also expect  ParameterResolver creation to be centralized here.